src/config.ts

```diff
@@ -53,13 +53,22 @@ const env = envalid.cleanEnv(process.env, {
     JOBS_CREATION_LOCK_TTL_MS: envalid.num({ default: 30000 }), // job creation lock ensures only one instance at a time can produce jobs
     SANITY_LOOP_PROCESSING_TIMEOUT_MS: envalid.num({ default: 180000 }), // max time allowed for a sanity job to finish processing until it times out - in ms
     METRICS_LOOP_INTERVAL_MS: envalid.num({ default: 60000 }), // time in ms
-    OCI_CONFIGURATION_FILE_PATH: envalid.str(),
-    OCI_CONFIGURATION_PROFILE: envalid.str({ default: 'DEFAULT' }),
     REPORT_EXT_CALL_MAX_TIME_IN_SECONDS: envalid.num({ default: 60 }),
     REPORT_EXT_CALL_MAX_DELAY_IN_SECONDS: envalid.num({ default: 30 }),
     REPORT_EXT_CALL_RETRYABLE_STATUS_CODES: envalid.str({ default: '429 409' }), // Retry on Too Many Requests, Conflict
+    CLOUD_PROVIDER: envalid.str({ default: 'oracle' }),
+    OCI_CONFIGURATION_FILE_PATH: envalid.str({ default: '' }),
+    OCI_CONFIGURATION_PROFILE: envalid.str({ default: '' }),
 });
 
+if (env.CLOUD_PROVIDER === 'oracle') {
```

Comment:
In other places we're setting the cloud provider as a property of the group rather than the whole process.  I understand why we'd want it this way for now, but eventually we could consider a mode wherein this service could manage multiple clouds at once as well.  So maybe we could consider a DEFAULT_CLOUD_PROVIDER setting here instead, and then not require any specific cloud provider config until later during instance launch.

Not a blocker, just a note for discussion.

Reply:
I like that all configuration check is done when launching the service, and not _later_.

But I agree, we could either read all `cloud` properties from groups or having a new env variable CLOUD_PROVIDERS provided as an array or with comma separated values